Lib/test/_test_multiprocessing.py

```diff
@@ -5849,6 +5901,24 @@ def test__all__(self):
         support.check__all__(self, multiprocessing, extra=multiprocessing.__all__,
                              not_exported=['SUBDEBUG', 'SUBWARNING'])
 
+    @only_run_in_spawn_testsuite("avoids redundant testing.")
```

Comment:
Why are the two skip messages different? Aren't they both spawn-specific, and wouldn't it be redundant to run them for other start methods in both cases?

(I wonder if it wouldn't make more sense to have these tests _not_ be in _test_multiprocessing, but in a separate test_multiprocessing. Saves you from the decorator and its test, too.)

Reply:
There's a big pile of refactoring in the way the entire multiprocessing test suites are setup that is needed.  The decorator is a start, I'd expect anything so-labelled with this decorator to be extracted out to a new non-all-start-methods file elsewhere if anyone ever takes that thankless task on.